lib/cadet/jobs/xml_parser.ex

```diff
@@ -169,6 +171,48 @@ defmodule Cadet.Updater.XMLParser do
     end
   end
 
+  defp process_voting(xml) do
+    voting_params =
+      xml
+      |> xpath(
+        ~x"//VOTING/VOTE"el,
+        entity: ~x"."
+      )
+      |> Enum.map(fn param ->
+        with vote when is_map(vote) <- process_vote(param),
+             vote when is_map(vote) <- Map.delete(vote, :entity) do
+          vote
+        else
+          {:error, error_message} -> {:error, error_message}
+        end
+      end)
+
+    if Enum.any?(voting_params, &(!is_map(&1))) do
+      error = Enum.find(voting_params, &(!is_map(&1)))
+      error
+    else
+      {:ok, voting_params}
+    end
+  end
+
+  defp process_vote(voting) do
+    vote =
+      voting[:entity]
+      |> xpath(
+        ~x"."e,
+        user_id: ~x"./@userid"oi,
+        submission_id: ~x"./@submissionid"oi
+      )
+
+    if is_map(vote) do
+      Logger.info(inspect(vote))
```

Comment:
Please remove this when you're done; log messages are fine but don't dump the entire struct

Reply:
has been removed